### PR TITLE
Fixes display issues with the new darkmode

### LINKED ISSUE #5394: Loading screen should be part of themes (to stop white loading screen melting eyeballs at night)
## The request

Currently, despite using a dark theme, if you load HA at night, the bright white loading screen melts eyeballs until frontend loads and a dark theme kicks in.

As requested, here is a picture of my eyeballs melting before my dark theme kicks in:

image

## The alternatives

Sunglasses?

### LINKED ISSUE #6528: Disabled button text color in Dark Mode (0.114)
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

## The problem

The new dark mode looks amazing, near perfect. The one thing I have a tiny nitpick on is the disabled button text colors. It's black on black (or a very dark gray), so makes it very hard to notice they're even there. See below screenshot for an example, you can barely see the bottom button row.

image

## Expected behavior

I think these should be a more lighter shade of gray, so you know they're there but disabled.

## Steps to reproduce

1. Enable dark mode
2. Bring up a dialog with disabled button (Users panel)

## Environment

- Home Assistant release with the issue: 0.114.0b0
- Last working Home Assistant release (if known): n/a
- Browser and browser version: Chrome
- Operating system: macOS

## State of relevant entities

```yaml

```

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

```txt

```

### LINKED ISSUE #6529: 2FA Dialog on Dark Mode (0.114.0b0)
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem

When using the new dark mode, you can not scan the QR code in the 2FA/TOTP dialog. This is an identical issue to the one I opened last year https://github.com/home-assistant/frontend/issues/4104 and it was fixed in https://github.com/home-assistant/frontend/pull/4107 by adding/forcing a white background to it. Seems like that background got lost with all the theme changes in 0.114 though and should be added back.

image

## Environment

- Home Assistant release with the issue: 0.114.0b0
- Last working Home Assistant release (if known): 0.113
- Browser and browser version: Chrome
- Operating system: macOS

## State of relevant entities

```yaml

```

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

```txt

```

## Additional information
QR Codes are used elsewhere too, I know that HomeKit setup/pairing is one of them but I am not able to test that. cc @bdraco

### LINKED ISSUE #6531: Integrations documentation icon on Dark Mode (0.114.0b0)
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem

On the integrations overflow menu where you can see a link to documentation, the "open in new / external link" icon appears to be hardcoded black. So on dark themes and the new dark mode, it's very difficult to see it's even there.

image

## Expected behavior

Visible icon

## Steps to reproduce

1. Enable Dark Mode
2. Go to Configuration -> Integrations
3. Click on the 3 dots / overflow menu for any integration
4. Look at the "Documentation" menu item

## Environment

- Home Assistant release with the issue: 0.114.0b0
- Last working Home Assistant release (if known): n/a
- Browser and browser version: Chrome
- Operating system: macOS

## State of relevant entities

```yaml

```

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

```txt

```

### PR DESCRIPTION
**Pull request by ludeeus**
### fixes #6531 

_Forces white background and black "text" for the svg_

image

### fixes #6529

_Uses a lighter color for the icon so it's easier to see_

image

### fixes #6528

_Uses a lighter color for the lebal so it's easier to see_

image

### closes #5394

_Set the backgrounds to dark in init (before loading) if dark is selected on the client_

image

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: 
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/integrations/ha-integration-card.ts [hunk 1]
```diff
@@ -453,6 +453,9 @@ export class HaIntegrationCard extends LitElement {
           cursor: pointer;
           min-height: 35px;
         }
+        ha-svg-icon {
```
Comment (bramkragten) [inline src/panels/config/integrations/ha-integration-card.ts:456]: You only want to target the more info icon right? Now you also target the overflow icon.
```suggestion
        mwc-list-item ha-svg-icon {
```

Comment (ludeeus) [reply src/panels/config/integrations/ha-integration-card.ts:456]: I wanted both, to ensure the theme, but the overflow is this color already so it does not really matter

File: src/resources/styles.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ export const darkStyles = {
   "switch-unchecked-button-color": "#999999",
   "switch-unchecked-track-color": "#9b9b9b",
   "divider-color": "rgba(225, 225, 225, .12)",
+  "mdc-button-disabled-ink-color": "rgba(225, 225, 225, .30)",
```
Comment (bramkragten) [inline src/resources/styles.ts:14]:
```suggestion
  "disabled-text-color": "rgba(225, 225, 225, .30)",
```
And if that doesn't yet, we should add
```
  "mdc-button-disabled-ink-color": "disabled-text-color",
```
to derived styles.

**Review by bramkragten — APPROVED** [review]

File: src/resources/styles.ts [hunk 3]
```diff
@@ -66,6 +65,7 @@ export const derivedStyles = {
   "mdc-radio-unchecked-color": "rgba(var(--rgb-primary-text-color), 0.54)",
   "mdc-radio-disabled-color": "var(--disabled-text-color)",
   "mdc-tab-text-label-color-default": "var(--primary-text-color)",
+  "mdc-button-disabled-ink-color": "disabled-text-color",
```
Comment (bramkragten) [inline src/resources/styles.ts:68]:
```suggestion
  "mdc-button-disabled-ink-color": "var(--disabled-text-color)",
```

**Comment by bdraco** [comment]
@SeanPM5 

Screen Shot 2020-08-06 at 10 57 34 AM

**Comment by bdraco** [comment]
Fixed homekit here https://github.com/home-assistant/core/pull/38609

**Pull request merged**